lib/zclient.c

```diff
@@ -1124,6 +1124,11 @@ int zapi_srv6_locator_encode(struct stream *s, const struct srv6_locator *l)
 	stream_put(s, l->name, strlen(l->name));
 	stream_putw(s, l->prefix.prefixlen);
 	stream_put(s, &l->prefix.prefix, sizeof(l->prefix.prefix));
+	stream_putc(s, l->block_bits_length);
+	stream_putc(s, l->node_bits_length);
+	stream_putc(s, l->function_bits_length);
+	stream_putc(s, l->argument_bits_length);
+	stream_putc(s, l->flags);
 	return 0;
```

Comment:
zebra uses these encode/decode functions to communicate to the locator information to protocols (BGP, ISIS, ...).

zebra needs to provide also block/node/func lengths, because protocols must advertise them in the corresponding TLVs

That's the reason why we need to include block/node/func lengths in the encode/decode functions.